### PR TITLE
feat(ENG-1380): task chain demo project

### PR DESCRIPTION
**Pull request by daabr**
This project demonstrates 3 execution approaches side-by-side: a single workflow that includes retries ("basic" + "advanced" modes), and event-driven workflows.

Refs: ENG-1380

### CODE REVIEW

File: task_chain/autokitteh.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# This YAML file is a declarative manifest that describes the setup
+# of an AutoKitteh project that ...TODO...
```
Comment (itayd) [inline task_chain/autokitteh.yaml:2]: TODO

**Comment by itayd** [comment]
add README

**Comment by itayd** [comment]
1. split to parallel subdirs to show different ways to tackle this
2. simplify

File: task_chain/autokitteh.yaml (hunk 1 above)
Comment (daabr) [reply task_chain/autokitteh.yaml:2]: Ack, already noted in the PR description. I'll do this (and a README) when the code structure is acceptable.

**Review by haimzlato — APPROVED** [review]
Looks good

**Pull request merged**